hedyorg/hedy
# Ru student details overhaul

[To be reviewed and approved by members of the NML team]

The student details section now shows data that is linked to the actual database (apart from the chart which will be a separate issue). Moreover, the codebase was overhauled in several aspects to avoid duplication and overall correctness. 

Relevant technical choices:
- Program error is retrieved from database and run through a new function which compiles the error message to the code
- Routing arguments such as which panels to show/collapse are generated on the fly in the backend and passed to the front end. This is so there is less code duplication in the html file and adding/removing/changing args is completely dynamic. 
- While there is some logic that passed data to the chart it doesn't work yet. This will be tackled in the next issue

How to test:
- Open instance of hedy on this branch where you log in as teacher
- Open instance of hedy on this branch where you log in as a student (incognito tabs work well)
- Go to an adventure as a student and make a deliberate run where you get an error
- See the error show up in the dashboard as a teacher, along with the code of that error
